### ISSUE TITLE
feature: fix hard code and make install more friendly

### ISSUE THREAD

**Issue by hustjieke** [stoneatom/stonedb#336]
**Is your feature request related to a problem? Please describe.**

Fix hard code on and more friendly
in install_scripts/CMakeList.txt
```
SET (INSTALL_ENGINE "/stonedb57")
SET (ENGINE_MYSQLDIR   "${INSTALL_ENGINE}/install")
install(FILES my.cnf
              DESTINATION ${ENGINE_MYSQLDIR} COMPONENT storage-engine)
install(PROGRAMS mysql_server install.sh reinstall.sh
              DESTINATION ${ENGINE_MYSQLDIR} COMPONENT storage-engine)
```

in storage/tianmu/CMakeLists.txt
```
#lihao Here, the dependency libs will be moved to 'third_party' directory.
SET(BOOST_ROOT ${WITH_BOOST})
SET(MARISA_ROOT /usr/local/stonedb-marisa)
SET(ROCKSDB_ROOT /usr/local/stonedb-gcc-rocksdb)
```

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by hustjieke** [comment]
Add variable defined in cmake,  like:
```
# begin to build
cmake ../../ \
-DCMAKE_BUILD_TYPE=${build_type} \
-DCMAKE_INSTALL_PREFIX=${install_target} \
-DMYSQL_DATADIR=${install_target}/data \
-DSYSCONFDIR=${install_target} \
-DMYSQL_UNIX_ADDR=${install_target}/tmp/mysql.sock \
-DWITH_EMBEDDED_SERVER=OFF \
-DWITH_STONEDB_STORAGE_ENGINE=1 \
-DWITH_MYISAM_STORAGE_ENGINE=1 \
-DWITH_INNOBASE_STORAGE_ENGINE=1 \
-DWITH_PARTITION_STORAGE_ENGINE=1 \
...
...
```

**Comment by hustjieke** [comment]
```
#lihao Here, the dependency libs will be moved to 'third_party' directory.
SET(BOOST_ROOT ${WITH_BOOST})
SET(MARISA_ROOT /usr/local/stonedb-marisa)
SET(ROCKSDB_ROOT /usr/local/stonedb-gcc-rocksdb)
```
had been resolved in #11
